pep-0458-tuf-online-keys.rst

```diff
@@ -499,16 +701,143 @@ SHOULD also be made available to clients.
 PyPI and Key Requirements
 =========================
 
-In this section, the kinds of keys required to sign for TUF roles on PyPI are
-examined.  TUF is agnostic with respect to choices of digital signature
-algorithms.  For the purpose of discussion, it is assumed that most digital
-signatures will be produced with the well-tested and tried RSA algorithm [20]_.
-Nevertheless, we do NOT recommend any particular digital signature algorithm in
-this PEP because there are a few important constraints: first, cryptography
-changes over time; second, package managers such as pip may wish to perform
-signature verification in Python, without resorting to a compiled C library, in
-order to be able to run on as many systems as Python supports; and third, TUF
-recommends diversity of keys for certain applications.
+The number of keys, and key type, required to sign the TUF metadata are
+discussed in this section.  TUF is agnostic with respect to the digital
+signature algorithms allowed to sign the TUF metadata, however, the Ed25519
+signature scheme [25]_ SHOULD be used by PyPI administrators.
+
+Ed25519 is a public-key signature system that uses small cryptographic
+signatures and keys.  It is an elliptic curve digital signature algorithm based
+on Twisted Edwards curves.  A pure-Python implementation [27]_ of the Ed25519
+signature scheme is available, and verification of Ed25519 signatures is fast
+even when performed in Python.
+
+The package manager (pip) shipped with CPython MUST work on non-CPython
+interpreters and cannot have dependencies that have to be compiled (i.e., the
+PyPI + TUF integration MUST NOT require compilation of C extensions in order to
+verify cryptographic signatures). Verification of signatures MUST be done in
+Python, and verifying RSA [20]_ signatures in pure-Python may be impractical
+due to speed. Therefore, PyPI SHOULD use the Ed25519 signature scheme.
+
+
+Cryptographic Key Files
+-----------------------
+
+Cryptographic keys MAY be stored in password-protected, encrypted key files.
+Administrators MAY use the repository tool to encrypt key files with
+AES-256-CTR-Mode and strengthen passwords with PBKDF2-HMAC-SHA256 (100K
+iterations by default, but this may be overridden by the developer). The
+current Python implementation of TUF can use any cryptographic library
+(PyCrypto [24]_ is currently used to encrypt the TUF key files, but support for
+PyCA Cryptography can be added in the future), may override the default number
+of PBKDF2 iterations, and the KDF tweaked to preference.  However, the exact
+cryptographic constructions can be adjusted to include future primitives added
+by the cryptographic libraries supported.
```

Comment:
735: something is wrong with "supported" - at the very least it is in the wrong place.